frontend/src/app/pages/exercises/exercise/shared/exercise-map/utility/ol-map-interactions-manager.ts

```diff
@@ -0,0 +1,173 @@
+import type VectorLayer from 'ol/layer/Vector';
+import type VectorSource from 'ol/source/Vector';
+import { selectCurrentRole } from 'src/app/state/application/selectors/shared.selectors';
+import type { Interaction } from 'ol/interaction';
+import { defaults as defaultInteractions } from 'ol/interaction';
+import type { Subject } from 'rxjs';
+import { combineLatest, takeUntil } from 'rxjs';
+import { selectExerciseStatus } from 'src/app/state/application/selectors/exercise.selectors';
+import type { Feature } from 'ol';
+import { Collection } from 'ol';
+import type OlMap from 'ol/Map';
+import type { AppState } from 'src/app/state/app.state';
+import type { Store } from '@ngrx/store';
+import { selectStateSnapshot } from 'src/app/state/get-state-snapshot';
+import type { ExerciseStatus, Role } from 'digital-fuesim-manv-shared';
+import { TranslateInteraction } from './translate-interaction';
+import type { PopupManager } from './popup-manager';
+import type { FeatureManager } from './feature-manager';
+
+export class OlMapInteractionsManager {
+    private readonly featureLayers: VectorLayer<VectorSource>[];
+    private readonly trainerInteractions: Interaction[];
+    private translateInteraction: TranslateInteraction;
+    private participantInteractions: Interaction[];
+    private interactions: Collection<Interaction>;
+    private lastStatus: ExerciseStatus | undefined;
+    private lastRole: Role | 'timeTravel' | undefined;
+
+    constructor(
+        private readonly mapInteractions: Collection<Interaction>,
+        private readonly store: Store<AppState>,
+        private readonly popupManager: PopupManager,
+        private readonly olMap: OlMap,
+        private readonly layerFeatureManagerDictionary: Map<
+            VectorLayer<VectorSource>,
+            FeatureManager<any>
+        >,
+        private readonly destroy$: Subject<void>
+    ) {
+        this.featureLayers = [];
+        this.trainerInteractions = [];
+        this.interactions = new Collection<Interaction>();
+        this.translateInteraction = new TranslateInteraction();
+        this.participantInteractions = [this.translateInteraction];
+        this.updateInteractions();
+        this.registerInteractionEnablementHandler();
+    }
+
+    public addFeatureLayer(layer: VectorLayer<VectorSource>) {
+        this.featureLayers.push(layer);
+        this.updateRegisterAndApplyAll();
+    }
+
+    public addTrainerInteraction(interaction: Interaction) {
+        this.trainerInteractions.push(interaction);
+        this.updateRegisterAndApplyAll();
+    }
+
+    private updateRegisterAndApplyAll() {
```

Comment:
I don't like the naming of this method. I sounds like you are updating the registration, because the key word `interaction` is missing. Do you mind adding it to the name?

Reply:
Suggestion: `syncInteractions`

Reply:
Done